rotkehlchen/tasks/calendar.py

```diff
@@ -47,3 +70,96 @@ def delete_past_calendar_entries(database: DBHandler) -> None:
+                    event_subtypes=[event_subtype],
+                ),
+            ))
+
+    with database.conn.read_ctx() as cursor:
+        blockchain_accounts = database.get_blockchain_accounts(cursor=cursor)
+
+    current_ts = ts_now()
+    ens_to_event: dict[str, tuple[int, EvmEvent]] = {}
+    for ens_event in ens_events:
+        if not (
+            ens_event.location_label is not None and
+            ens_event.extra_data is not None and
+            'fullname' in ens_event.extra_data and
+            'expires' in ens_event.extra_data and (
+                user_address := string_to_evm_address(ens_event.location_label)
+            ) in blockchain_accounts.get(
+                blockchain := ChainID.deserialize(ens_event.location.to_chain_id()).to_blockchain(),  # noqa: E501
+            ) and (ens_expires := Timestamp(ens_event.extra_data['expires'])) > current_ts  # those with expiry in the future  # noqa: E501
+        ):
+            continue
+
+        ens_name: str = ens_event.extra_data['fullname']
+        if (calendar_entries := db_calendar.query_calendar_entry(
+            filter_query=CalendarFilterQuery.make(
+                and_op=True,
+                name=f'{ens_name} expiry',
+                addresses=[OptionalBlockchainAddress(
+                    blockchain=blockchain,
+                    address=user_address,
+                )],
+                blockchain=blockchain,
+                counterparty=CPT_ENS,
+            ),
+        ))['entries_found'] == 0:  # if calendar entry doesn't exist, add it
+            calendar_entry_id = db_calendar.create_calendar_entry(CalendarEntry(
+                name=f'{ens_name} expiry',
+                timestamp=ens_expires,
+                description=f'{ens_name} expires on {customizable_date.timestamp_to_date(ens_expires)}',  # noqa: E501
+                counterparty=CPT_ENS,
+                address=user_address,
+                blockchain=blockchain,
+                color=None,
+                auto_delete=True,
+            ))
+        else:  # else
+            calendar_entry_id = calendar_entries['entries'][0].identifier
+
+        if (
+            ens_name not in ens_to_event or
+            ens_expires > ens_to_event[ens_name][1].extra_data['expires']  # type: ignore[index]  # extra_data is not None, checked above
+        ):  # insert mapping for the latest expiry timestamp
+            ens_to_event[ens_name] = (calendar_entry_id, ens_event)
+
+    _, failed_to_add = db_calendar.create_reminder_enties(reminders=[
+        ReminderEntry(
+            identifier=idx,  # this is only used for logging below, it's auto generated in db
+            event_id=calendar_identifier,
+            secs_before=secs_before,
+        )
+        for idx, (calendar_identifier, _) in enumerate(ens_to_event.values())
+        if len(db_calendar.query_reminder_entry(event_id=calendar_identifier)['entries']) == 0  # if not already present  # noqa: E501
```

Comment:
I would do just a count since here you are doing deserializations that you don't need if you are just counting

Reply:
Also shoudn't you check if the `event_id=calendar_identifier` but also if secs_before is the amount that you are adding the reminder for? With the logic as it is you could be adding one entry and then the second won't be added because you added already one and the count == 1

Reply:
I did not check the `secs_before` because it's possible that the user could have already set their own custom reminder, in that case, we do not add it. Also, the count should not increase because it's a list comprehension before the full query, and then all of them are added at once.